packages/apollo-gateway/src/index.ts

```diff
@@ -300,12 +300,30 @@ export class ApolloGateway implements GraphQLService {
   }
 
   public async load(options?: { engine?: GraphQLServiceEngineConfig }) {
-    await this.updateComposition(options);
+    if (options && options.engine) {
+      if (!options.engine.graphVariant)
+        this.logger.warn('No graph variant provided. Defaulting to `current`.');
+      this.engineConfig = options.engine;
+    }
 
     if (isManagedConfig(this.config) || this.experimental_pollInterval) {
+      await this.updateComposition();
       if (!this.pollingTimer) this.pollServices();
+    } else {
+      try {
+        await this.updateComposition();
+      } catch {
+        // In the case that we're neither managed nor polling, the gateway should
+        // crash in the event that it can't load service definitions. Leaving the
+        // gateway running in a non-operable state doesn't make sense and this
+        // allows container managers like K8s to restart the broken container.
+        // Note: the error caught within `this.updateComposition` handles logging
```

Comment:
Curious: What's the justification for having the error printed in `updateComposition` rather than here?   There is fair amount of functionality that seems like it could `throw` in there and, broadly, this seems like a better place to catch any of those errors that are not explicitly `catch`'d within `updateComposition` right now (e.g., `queryPlanStore.flush`-ing, invocations of `experimental_didUpdateComposition`, etc.)

Reply:
No longer applies / resolved by afa6746